### PR TITLE
WIP for hosting nldas

### PR DESCRIPTION
**Pull request by jread-usgs**
Have a few things I just noticed that I will make notes on. Also, this had some old commits mixed in :\

### CODE REVIEW

File: p1_import/out/nldas.ncml [hunk 1]
```diff
@@ -0,0 +1,192 @@
+<netcdf xmlns="http://www.unidata.ucar.edu/namespaces/netcdf/ncml-2.2" xmlns:xlink="http://www.w3.org/1999/xlink">
+  <aggregation type="union">
+    <netcdf>
+      <aggregation type="joinExisting" dimName="time">
+        <netcdf location="NLDAS_325675.326387_0.223_0.463_pressfc.nc"/>
```
Comment (jread-usgs) [inline p1_import/out/nldas.ncml]: this needs to sort in the opposite order.

File: remake_nldas.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+packages:
```
Comment (jread-usgs) [inline remake_nldas.yaml]: basic remake structure @aappling-usgs

File: p1_import/code/00_subset_nldas.R [hunk 3]
```diff
@@ -0,0 +1,143 @@
+
+
+nc_sub_grids <- function(nldas_config){
```
Comment (jread-usgs) [inline p1_import/code/00_subset_nldas.R]: All the functions in the nldas remake. Could separate this into other files if desired.

Comment (aappling-usgs) [inline p1_import/code/00_subset_nldas.R]: what does nc stand for?

File: configs/nldas_subset.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+data_name: "NLDAS"
+sub_times: ["2006-01-01 UTC", "2016-03-27 UTC"]
```
Comment (aappling-usgs) [inline configs/nldas_subset.yaml]: set to 2006-10-01 start time?

Comment (jread-usgs) [inline configs/nldas_subset.yaml]: figured I would be inclusive in case we wanted to go earlier

File: p1_import/code/00_subset_nldas.R (hunk 3 above)
Comment (jread-usgs) [inline p1_import/code/00_subset_nldas.R]: .nc is for netcdf files

File: README.Rmd [hunk 5]
```diff
@@ -1,5 +1,27 @@
...
+
+```{r}
+install.packages(
+ c("dataRetrieval", "R.utils", "geoknife"), 
+ repos=c("http://owi.usgs.gov/R","http://cran.us.r-project.org"),
+ dependencies=TRUE,
+ type="both")
+```
+
+The workflow is documented and enforced in 'makefile', which can be run with a call to `make` from the command line, or using Ctrl-Shift-B in RStudio. At least, we would like this to be true; in practice, there's lots of babysitting of loops through web API calls and computing clusters. Still, the makefile should give you an idea of how we want things to flow.
```
Comment (aappling-usgs) [inline README.Rmd]: :+1: to all the readme info

File: gap_analysis/long_term_sites_analysis.R [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+site.loc <- select(readNWISsite(all_sites$site_no), dec_lat_va, dec_long_va, site_no)
+site.points <- data.frame(lon=site.loc$dec_long_va, lat = site.loc$dec_lat_va)
+sites.proj <- SpatialPoints(site.points, proj4string=CRS("+proj=longlat + datum=wgs84")) %>% 
+  spTransform(CRS(plot.CRS))
+
+
+
+
+png(filename = 'long_term_sites.png', width = 5.2, height=3.5, res=200, units = 'in')
```
Comment (aappling-usgs) [inline gap_analysis/long_term_sites_analysis.R]: maybe write this png and the next table to somewhere within the gap_analysis directory?

File: p1_import/code/00_subset_nldas.R [hunk 7]
```diff
@@ -0,0 +1,143 @@
+
```
Comment (aappling-usgs) [inline p1_import/code/00_subset_nldas.R]: could you slap a Roxygen description line on each of these functions while they're fresh in your mind, please? i know it's your domain right now, but will help me in jumping into this stuff later if needed.

**Comment by jread-usgs** [comment]
will amend. Thanks

**Pull request merged**